Sources/tart/OCI/Registry.swift

```diff
@@ -104,13 +104,11 @@ class Registry {
   convenience init(
     host: String,
     namespace: String,
+    insecure: Bool = false,
     credentialsProvider: CredentialsProvider = KeychainCredentialsProvider()
   ) throws {
-    var baseURLComponents = URLComponents()
-
-    baseURLComponents.scheme = "https"
-    baseURLComponents.host = host
-    baseURLComponents.path = "/v2/"
+    let proto = insecure ? "http" : "https"
+    let baseURLComponents = URLComponents(string: proto + "://" + host + "/v2/")!
```

Comment:
Can we just assume that post might contain either `https` or `http` and do something like:

```swift
var baseURL = host + "/v2/"
if !baseURL.contains("://") {
  baseURL = "https://" + baseURL
}
```

Will it be properly parsed from the upstream invocations?

Reply:
We'd have to modify the OCI reference parser (`RemoteName.swift`) for this. And an OCI reference parser that accepts URLs and knows anything about protocol seems like a standard non-compliant thing that might bite us in the future.

It also seems weird once we allow `https://ghcr.io/a/b` syntax instead of `ghcr.io/a/b`, it might make someone think they can control the base URL, which this is not the case: the base URL will ignore `/a/b, but will always include `/v2/`.